File: src/main/java/br/com/marconardes/storyflame/swing/model/Chapter.java
package br.com.marconardes.storyflame.swing.model;

import java.util.UUID;

public class Chapter {
    private String id;
    private String title;
    private int order;
    private String content;
    private String summary;

    // Constructor
    public Chapter(String title, int order) {
        this.id = UUID.randomUUID().toString();
        this.title = title;
        this.order = order;
        this.content = "";
        this.summary = "";
    }

    // Getters and Setters
    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public int getOrder() {
        return order;
    }

    public void setOrder(int order) {
        this.order = order;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public String getSummary() {
        return summary;
    }

    public void setSummary(String summary) {
        this.summary = summary;
    }

    @Override
    public String toString() {
        return title; // For display in JList or similar components
    }
}

File: composeApp/src/commonMain/kotlin/br/com/marconardes/model/Chapter.kt
package br.com.marconardes.model

import com.benasher44.uuid.uuid4
import kotlinx.serialization.Serializable

@Serializable
data class Chapter(
    val id: String = uuid4().toString(),
    val title: String,
    val order: Int, // Assuming manual order for now, can be creationTimestamp or similar
    val content: String = "", // Actual text content of the chapter
    val summary: String = "" // New property for chapter summary
)
